fs/tmpfs/fs_tmpfs.c

```diff
@@ -455,6 +455,13 @@ static int tmpfs_find_dirent(FAR struct tmpfs_directory_s *tdo,
 {
   int i;
 
+  if (len && name[len - 1] == '/')
```

Comment:
Why don't you do the same as you did for namelen:
if (len == 0)
  {
    return -EINVAL;
  }
else if (name[len - 1] == '/')

Reply:
Done.